core/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -342,17 +411,21 @@ private void parseSub(XContentParser parser, FieldParser fieldParser, String cur
 
     private FieldParser getParser(String fieldName) {
         FieldParser parser = fieldParserMap.get(fieldName);
-        if (parser == null && false == ignoreUnknownFields) {
-            throw new IllegalArgumentException("[" + name  + "] unknown field [" + fieldName + "], parser not found");
+        if (parser == null) {
+            if (this.unknownFieldParser != null) {
+                parser = this.unknownFieldParser;
+            } else if (false == this.ignoreUnknownFields) {
+                throw new IllegalArgumentException("[" + name  + "] unknown field [" + fieldName + "], parser not found");
```

Comment:
Yes, I will think about this again in the context of #24020 which should go in first.